code/modules/modular_computers/computers/modular_computer/interaction.dm

```diff
@@ -0,0 +1,300 @@
+		update_verbs()
+		playsound(loc, 'sound/machines/id_swipe.ogg', 100, 1)
+		to_chat(user, "You insert [I] into [src].")
+
+		return
+	if(istype(W, /obj/item/weapon/pen) && stores_pen)
+		if(istype(stored_pen))
+			to_chat(user, "<span class='notice'>There is already a pen in [src].</span>")
+			return
+		if(!user.unEquip(W, src))
+			return
+		stored_pen = W
+		update_verbs()
+		to_chat(user, "<span class='notice'>You insert [W] into [src].</span>")
+		return
+	if(istype(W, /obj/item/weapon/paper))
+		var/obj/item/weapon/paper/paper = W
+		if(scanner && paper.info)
+			scanner.do_on_attackby(user, W)
+			return
+	if(istype(W, /obj/item/weapon/paper) || istype(W, /obj/item/weapon/paper_bundle))
+		if(nano_printer)
+			nano_printer.attackby(W, user)
+	if(istype(W, /obj/item/device/aicard))
+		if(!ai_slot)
+			return
+		ai_slot.attackby(W, user)
+
+	if(!modifiable)
+		return ..()
+
+	if(istype(W, /obj/item/weapon/computer_hardware))
+		var/obj/item/weapon/computer_hardware/C = W
+		if(C.hardware_size <= max_hardware_size)
+			try_install_component(user, C)
+		else
+			to_chat(user, "This component is too large for \the [src].")
+
+	var/list/usable_qualities = list(QUALITY_SCREW_DRIVING, QUALITY_WELDING, QUALITY_BOLT_TURNING)
+
+	var/tool_type = W.get_tool_type(user, usable_qualities)
+	switch(tool_type)
+		if(QUALITY_BOLT_TURNING)
+			var/list/components = get_all_components()
+			if(components.len)
+				to_chat(user, "Remove all components from \the [src] before disassembling it.")
+				return
+			if(W.use_tool(user, src, WORKTIME_SLOW, QUALITY_BOLT_TURNING, FAILCHANCE_VERY_EASY, required_stat = STAT_COG))
+				new /obj/item/stack/material/steel( get_turf(src.loc), steel_sheet_cost )
+				src.visible_message("\The [src] has been disassembled by [user].")
+				qdel(src)
+				return
+
+		if(QUALITY_WELDING)
+			if(!damage)
+				to_chat(user, "\The [src] does not require repairs.")
+				return
+			if(W.use_tool(user, src, WORKTIME_SLOW, QUALITY_WELDING, FAILCHANCE_HARD, required_stat = STAT_COG))
+				damage = 0
+				to_chat(user, "You repair \the [src].")
+				return
+
+		if(QUALITY_SCREW_DRIVING)
+			var/list/all_components = get_all_components()
+			if(!all_components.len)
+				to_chat(user, "This device doesn't have any components installed.")
+				return
+			var/list/component_names = list()
+			for(var/obj/item/weapon/computer_hardware/H in all_components)
+				component_names.Add(H.name)
+			var/choice = input(usr, "Which component do you want to uninstall?", "Computer maintenance", null) as null|anything in component_names
+			if(!choice)
+				return
+			if(!Adjacent(usr))
+				return
+			if(W.use_tool(user, src, WORKTIME_FAST, QUALITY_SCREW_DRIVING, FAILCHANCE_VERY_EASY, required_stat = STAT_COG))
+				var/obj/item/weapon/computer_hardware/H = find_hardware_by_name(choice)
+				if(!H)
+					return
+				uninstall_component(user, H)
+				return
+	..()
+
+/obj/item/modular_computer/examine(var/mob/user)
+	. = ..()
+
+	if(enabled && .)
```

Comment:
`.` return TRUE if distance between `user` and `src` is lower then second arg what was passed to `..()`
Will it work here?

Reply:
Default argument is -1 thats means no distance check at all